src/pybind/mgr/cephadm/services/monitoring.py

```
@@ -85,7 +86,15 @@ class AlertmanagerService(CephadmService):
 
     def prepare_create(self, daemon_spec: CephadmDaemonSpec[AlertManagerSpec]) -> CephadmDaemonSpec:
         assert self.TYPE == daemon_spec.daemon_type
+        # if spec is not attached to daemon_spec it is likely a redeploy or reconfig and
+        # spec should be in spec store
```

can we please fail hard, if we don't have a spec?